File: backend/FleetMind.Api/Models/Incident.cs
using System;
using FleetMind.Api.Models.Common;

namespace FleetMind.Api.Models
{
    public class Incident : BaseEntity
    {
        public Guid ShipId { get; set; }
        public Ship Ship { get; set; } = null!;

        public Guid? VoyageId { get; set; }
        public Voyage? Voyage { get; set; }

        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Severity { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty; // Default set in config or DTO

        public Guid ReportedByUserId { get; set; }
        public User ReportedByUser { get; set; } = null!;

        public DateTime OccurredAt { get; set; }
        public DateTime? ResolvedAt { get; set; }
        public string? ResolutionNotes { get; set; }
    }
}

File: frontend/src/types/incident.ts
import { PaginationQuery } from './pagination';

export interface Incident {
  id: string;
  shipId: string;
  shipName: string;
  voyageId: string | null;
  voyageNumber: string | null;
  title: string;
  description: string;
  severity: string;
  status: string;
  reportedByUserId: string;
  reportedByUserName: string;
  occurredAt: string;
  resolvedAt: string | null;
  resolutionNotes: string | null;
  createdAt: string;
}

export interface CreateIncidentPayload {
  shipId: string;
  voyageId?: string | null;
  title: string;
  description: string;
  severity: string;
  occurredAt: string;
}

export interface UpdateIncidentPayload {
  title: string;
  description: string;
  severity: string;
  resolutionNotes?: string | null;
}

export interface UpdateIncidentStatusPayload {
  status: string;
  resolvedAt?: string | null;
  resolutionNotes?: string | null;
}

export interface IncidentQuery extends PaginationQuery {
  shipId?: string;
  status?: string;
  severity?: string;
}
